jenkins/spark-tests.sh

```diff
@@ -263,6 +263,20 @@ run_pyarrow_tests() {
   ./run_pyspark_from_build.sh -m pyarrow_test --pyarrow_test
 }
 
+run_non_utc_time_zone_tests() {
+  # select one time zone according to date
+  non_utc_time_zones=("Asia/Shanghai" "Iran")
+  time_zones_length=${#non_utc_time_zones[@]}
+  current_date=$(date +%Y%m%d)
```

Comment:
nit: may use the `date +%u` to get day of week, so schedule is expected unlike depends on data number